File: backend/src/main/java/com/salehhafyane/ecommerce/dto/Purchase.java
package com.salehhafyane.ecommerce.dto;

import com.salehhafyane.ecommerce.entity.Address;
import com.salehhafyane.ecommerce.entity.Order;
import com.salehhafyane.ecommerce.entity.OrderItem;
import com.salehhafyane.ecommerce.entity.User;
import lombok.Data;


import java.util.Set;
@Data
public class Purchase {
    private Address address;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: frontend/src/app/common/purchase.ts
import {Customer} from "./customer";
import {Address} from "./address";
import {Order} from "./order";
import {OrderItem} from "./order-item";

export class Purchase {

  address:Address
  order:Order
  orderItems:OrderItem[]
  constructor(

     address:Address,
     order:Order,
     orderItems:OrderItem[]
  ) {

    this.address = address
    this.order = order
    this.orderItems = orderItems
  }
}
